symforce/slam/imu_preintegration/preintegrated_imu_measurements.cc

```diff
@@ -0,0 +1,68 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#include "./preintegrated_imu_measurements.h"
+
+#include <sym/factors/imu_manifold_preintegration_update.h>
+
+namespace sym {
+
+template <typename Scalar>
+PIMNoCovariance<Scalar>::PIMNoCovariance(const Vector3& accel_bias, const Vector3& gyro_bias)
+    : DR(),
+      Dv{Vector3::Zero()},
+      Dp{Vector3::Zero()},
+      DR_D_gyro_bias{Matrix3::Zero()},
+      Dv_D_gyro_bias{Matrix3::Zero()},
+      Dv_D_accel_bias{Matrix3::Zero()},
+      Dp_D_gyro_bias{Matrix3::Zero()},
+      Dp_D_accel_bias{Matrix3::Zero()},
+      accel_bias{accel_bias},
+      gyro_bias{gyro_bias},
+      integrated_dt{0.0} {}
+
+template <typename Scalar>
+PreintegratedImuMeasurements<Scalar>::PreintegratedImuMeasurements(const Vector3& accel_bias,
+                                                                   const Vector3& gyro_bias)
+    : pim(accel_bias, gyro_bias), covariance{Matrix9::Zero()} {}
+
+template <typename Scalar>
+void PreintegratedImuMeasurements<Scalar>::IntegrateMeasurement(
+    const Vector3& measured_accel, const Vector3& measured_gyro, const Matrix3& accel_cov,
+    const Matrix3& gyro_cov, const Scalar dt, const Scalar epsilon) {
+  Rot3<Scalar> new_DR;
+  Vector3 new_Dv;
+  Vector3 new_Dp;
+  Matrix9 new_covariance;
+  Matrix3 new_DR_D_gyro_bias;
+  Matrix3 new_Dv_D_gyro_bias;
+  Matrix3 new_Dv_D_accel_bias;
+  Matrix3 new_Dp_D_gyro_bias;
+  Matrix3 new_Dp_D_accel_bias;
```

Comment:
It almost seems like this set of things (maybe minus `new_covariance`) wants to be an LCM type generated from a python dataclass/Values, which would mean it could be one argument to this function (or the return value), and most of the copies below would just be assigning the LCM type, and some other things might be simpler as well.  Just an idea, not necessarily sold on it